### ISSUE TITLE
Dependency `encoding` is unmaintained

### ISSUE THREAD

**Issue by mati865** [adamcrume/java-properties#9]
Hello,
Cargo-deny shows warning when using this crate because the dependency `encoding` is unmaintained:
```
warning[unmaintained]: `encoding` is unmaintained
   ┌─ <project_path>/Cargo.lock:23:1
   │
23 │ encoding 0.2.33 registry+https://github.com/rust-lang/crates.io-index
   │ --------------------------------------------------------------------- unmaintained advisory detected
   │
   = ID: RUSTSEC-2021-0153
   = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0153
   = Last release was on 2016-08-28. The [issue](https://github.com/lifthrasiir/rust-encoding/issues/127) inquiring as to the status of the crate has gone unanswered by the maintainer.
     
     ## Possible alternatives
     
     - [encoding_rs](https://crates.io/crates/encoding_rs)
   = Announcement: https://github.com/lifthrasiir/rust-encoding/issues/127
   = Solution: No safe upgrade is available!
   = encoding v0.2.33
     └── java-properties v1.4.1
```

**Comment by adamcrume** [comment]
Thanks for pointing this out!  I'm working on it, but this requires a lot of code changes.

**Comment by adamcrume** [comment]
This is fixed by 1a6c1d44.  This is a breaking change, so I'll have to release a new major version.  I'm going to see if there are any other potentially breaking changes I'd like to make before releasing a new version.

**Comment by mati865** [comment]
Thank you.
